ansible/roles/machine/provision/tasks/main.yml

```diff
@@ -1,8 +1,13 @@
 ---
-- name: create test config file
-  template:
-    src: ipa-test-config.yaml
-    dest: /vagrant/ipa-test-config.yaml
+- block:
+    - name: get DNS server from resolv.conf
+      shell: cat /etc/resolv.conf | grep nameserver | head -n 1 | sed -e 's/nameserver \(.*\)/\1/'
```

Comment:
You can replace cat, grep, head and sed with awk:
```
shell: awk '$1 == "nameserver" {print $2; exit}' /etc/resolv.conf
```

Reply:
That looks better, thanks! :)